src/components/frame/Extensions/pages/partials/InstallButton.tsx

```diff
@@ -35,28 +58,136 @@ const InstallButton = ({ extensionData }: InstallButtonProps) => {
   const navigate = useNavigate();
 
   const isMobile = useMobile();
+  const isStakedExpenditureExtension =
+    extensionData.extensionId === Extension.StakedExpenditure;
+
+  const asyncFunction = useAsyncFunction({
+    submit: ActionTypes.EXTENSION_ENABLE,
+    error: ActionTypes.EXTENSION_ENABLE_ERROR,
+    success: ActionTypes.EXTENSION_ENABLE_SUCCESS,
+  });
+
+  const isExtensionInstalled =
+    extensionData && isInstalledExtensionData(extensionData);
+
+  useEffect(() => {
+    const handleEnableExtension = async () => {
+      setWaitingForActionConfirmation(true);
+      try {
+        setIsPolling(true);
+        const extensionRoles = isExtensionInstalled
+          ? getUserRolesForDomain(colony, extensionData.address, Id.RootDomain)
+          : [];
+        const missingPermissions = extensionData.neededColonyPermissions.filter(
+          (neededRole) => {
+            return !userHasRole(extensionRoles, neededRole);
+          },
+        );
+
+        const updatedExtensionData = {
+          ...extensionData,
+          missingColonyPermissions: missingPermissions,
```

Comment:
This should be already handled by `useExtensionData`. Is there some race condition that forces you to edit them again?

Reply:
I got an error that this field is missing